src/etc/inc/services.inc

```diff
@@ -1972,14 +1972,29 @@ function dyndnsCheckIP($int) {
 		if (stristr($gateways_status[$config['interfaces'][$int]['gateway']]['status'], "down")) {
 			return "down";
 		}
-		$hosttocheck = "http://checkip.dyndns.org";
+
+		if (is_array($config['checkipservices']['checkipservice'])) {
+			foreach ($config['checkipservices']['checkipservice'] as $i => $checkipservice) {
+				if (isset($checkipservice['enable'])) {
+					$url = $checkipservice['url'];
+					$username = $checkipservice['username'];
+					$password = $checkipservice['password'];
+					$verifysslpeer = isset($checkipservice['verifysslpear']) ? true : false;
+					break;
+				}
+			}
+		}
+
+		$hosttocheck = $url;
```

Comment:
If the user deletes all the "Check IP Services" then $url is going to be empty here.
Perhaps there could be a function that returns the array of the "default" DynDNS check IP settings. That could be used here in an else of the above "if", and also in services_checkip_edit.php around line 90, where those "default" settings are currently done.